### PR TITLE
Modify pypi PKG-INFO parse

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
Reference: #2943
Do not return 2 PackageData objects when there are PKG-INFO files.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by JonoYang** [comment]
@AyanSinhaMahapatra LGTM!

File: src/packagedcode/pypi.py [hunk 1]
```diff
@@ -270,11 +270,13 @@ class PythonSdistPkgInfoFile(BaseExtractedPythonLayout):
 
     @classmethod
     def parse(cls, location):
-        yield parse_metadata(
-            location=location,
-            datasource_id=cls.datasource_id,
-            package_type=cls.default_package_type,
-        )
+        if not PythonEggPkgInfoFile.is_datafile(location) and not\
```
Comment (pombredanne) [inline src/packagedcode/pypi.py]: Please avoid back-slashes for line continuations. Prefer instead using `()` parens.
```suggestion
        if (not PythonEggPkgInfoFile.is_datafile(location) and not
```

Comment (pombredanne) [reply src/packagedcode/pypi.py]: Also I think it would be best overriding is_datafile() for this

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks. See my nits!

File: src/packagedcode/pypi.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply src/packagedcode/pypi.py]: Made the requested changes.

**Review by pombredanne — COMMENTED** [review]
Great!

**Review by pombredanne — APPROVED** [review]

**Pull request merged**